tinygrad/shape/symbolic.py

```diff
@@ -173,12 +181,14 @@ def get_bounds(self) -> Tuple[int, int]:
     return (self.a.min*self.b, self.a.max*self.b) if self.b >= 0 else (self.a.max*self.b, self.a.min*self.b)
 
 class DivNode(OpNode):
+  def infer(self, symbols): return sym_infer(self.a, symbols) // sym_infer(self.b, symbols)
   def __floordiv__(self, b: int, _=False): return self.a//(self.b*b) # two divs is one div
   def get_bounds(self) -> Tuple[int, int]:
     assert self.a.min >= 0 and isinstance(self.b, int)
     return self.a.min//self.b, self.a.max//self.b
 
 class ModNode(OpNode):
+  def infer(self, symbols): return sym_infer(self.a, symbols) % sym_infer(self.b, symbols)
```

Comment:
What does `infer` do?

Reply:
`infer` takes `symbols` and infer the value of the node. Used to infer symbolic shape to int shape. Added some unit tests for it

Reply:
I don't really understand, I think there's a simpler way to think about this.